### PR TITLE
tests: add config resolving tests for ConfigGroup

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

tests

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
NA

**Summary**
- We didn't have tests for ConfigGroup so introduced them in https://github.com/webpack/webpack-cli/pull/1768, this PR adds more tests for config resolution and to ensure we're getting the expected output

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

**Review by evilebottnawi — APPROVED** [review]
Good job!

File: packages/webpack-cli/__tests__/ConfigGroup/webpack.promise.config.cjs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module.exports = () => {
```
Comment (evenstensberg) [inline packages/webpack-cli/__tests__/ConfigGroup/webpack.promise.config.cjs:1]: cjs uses export default

Comment (anshumanv) [reply packages/webpack-cli/__tests__/ConfigGroup/webpack.promise.config.cjs:1]: That's for mjs, cjs uses module.exports AFAIU

Comment (evilebottnawi) [reply packages/webpack-cli/__tests__/ConfigGroup/webpack.promise.config.cjs:1]: Yes, `export default` is `mjs`

Comment (evenstensberg) [reply packages/webpack-cli/__tests__/ConfigGroup/webpack.promise.config.cjs:1]: ah right, could we add one for mjs too?

Comment (anshumanv) [reply packages/webpack-cli/__tests__/ConfigGroup/webpack.promise.config.cjs:1]: mjs support is WIP here - https://github.com/webpack/webpack-cli/pull/1728

Will add for mjs after I finish that work 😄

**Review by evenstensberg — APPROVED** [review]

**Pull request merged**